File: 이것이 취업을 위한 코딩 테스트다/그리디/$1이_될_때까지/$1이_될_때까지.java
package 그리디.$1이_될_때까지;

import java.util.*;

public class $1이_될_때까지 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();
    int answer = 0;

    while (n != 1) {
      if (n % k == 0) {
        n /= k;
      } else {
        n--;
      }

      answer++;
    }

    System.out.println(answer);
  }
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/$1이_될_때까지/$1이_될_때까지.cpp
#include <iostream>
using namespace std;

int n, k;

int main() {
  cin >> n >> k;
  int answer = 0;

  while (n != 1) {
    if (n % k == 0) {
      n /= k;
    } else {
      n--;
    }
    answer++;
  }

  cout << answer;
}